js/src/dom/manipulator.spec.js

```diff
@@ -0,0 +1,219 @@
+      const div = fixtureEl.querySelector('div')
+      Manipulator.setDataAttribute(div, 'tEsT', 'value')
+      expect(div.getAttribute('data-test')).toEqual('value')
+    })
+  })
+
+  describe('getDataAttribute', () => {
+    it('should get data attribute', () => {
+      fixtureEl.innerHTML = '<div data-test="null" ></div>'
+      const div = fixtureEl.querySelector('div')
+      expect(Manipulator.getDataAttribute(div, 'test')).toBeNull()
+    })
+
+    it('should get data attribute in lower case', () => {
+      fixtureEl.innerHTML = '<div data-test="value" ></div>'
+      const div = fixtureEl.querySelector('div')
+      expect(Manipulator.getDataAttribute(div, 'tEsT')).toEqual('value')
+    })
+  })
+
+  describe('getDataAttributes', () => {
+    it('should return empty object for null', () => {
+      expect(Manipulator.getDataAttributes(null), {})
+    })
+
+    it('should return empty object for undefined', () => {
+      expect(Manipulator.getDataAttributes(), {})
+    })
+
+    it('should get all data attributes', () => {
+      fixtureEl.innerHTML = '<div data-test="js" data-test2="js2" ></div>'
+      const div = fixtureEl.querySelector('div')
+      expect(Manipulator.getDataAttributes(div)).toEqual({
+        test: 'js',
+        test2: 'js2'
+      })
+    })
+
+/* TODO: this test appears in the original test suite. What is it doing?
+    it('all tests together', () => {
+      let fixtureEl1 = getFixture()
+      let fixtureEl2 = getFixture()
+      let fixtureEl3 = getFixture()
+      fixtureEl1.innerHTML = '<div data-test="js" data-test2="js2" ></div>'
+      fixtureEl2.innerHTML = '<div data-test3="js" data-test4="js2" ></div>'
+      fixtureEl3.innerHTML = '<div attri="1" ></div>'
+      const div1 = fixtureEl1.querySelector('div')
+      const div2 = fixtureEl2.querySelector('div')
+      const div3 = fixtureEl3.querySelector('div')
+
+      expect(Manipulator.getDataAttributes(div1)).toEqual({
+        test: 'js',
+        test2: 'js2'
+      })
+
+      expect(Manipulator.getDataAttributes(null)).toEqual({})
+
+      var stub = sinon
+        .stub(Object, 'getOwnPropertyDescriptor')
+        .callsFake(function () {
+          return false
+        })
+
+      expect(Manipulator.getDataAttributes(div2)).toEqual({
+        test3: 'js',
+        test4: 'js2'
+      })
+
+      expect(Manipulator.getDataAttributes(div3)).toEqual({})
+
+      stub.restore()
+    })
+*/
+  })
+
+  describe('removeDataAttribute', () => {
+    it('should throw error if element is null', () => {
+      const incorrect = function() {
+        removeDataAttribute(null, 'key')
+      }
+      expect(incorrect).toThrow();
+    })
+
+    it('should throw error if element is undefined', () => {
+      const incorrect = function() {
+        removeDataAttribute(undefined, 'key')
+      }
+      expect(incorrect).toThrow();
+    })
+
+    it('should remove data attribute', () => {
+      fixtureEl.innerHTML = '<div data-key="value" ></div>'
```

Comment:
trailing space after data-key="value"

Reply:
I'll remove it. (I added it in the first place out of extra caution because there I saw weird behavior somewhere that didn't like `<div [something]/>` but worked with `<div [something] />`.)